### DISCUSSION TITLE
Photo Demon using Avalonia UI

### DISCUSSION THREAD

**Discussion by Mtothexmax** [tannerhelland/PhotoDemon#475] [Ideas]
I really like the tool Photo Demon but I wonder if it would be possible to port it to Avalonia. It is quite similar to WPF and it is cross Platform compatible, so one can also use it on Mac, Linux and run it as a WebApp.

One can also program in it using Visual Basic and the Xaml code is quite similar (maybe it can even be 100% the same)

**Comment by manfromarce** [comment]
I am not the developer but I think it would be too much work because PhotoDemon does not use XAML like WPF/Avalonia, maybe Eto Forms or twinBasic (which should be free for open source projects) would be a better choice.

**Comment by tannerhelland** [comment]
@manfromarce is exactly right.  (Thank you for the comment!)

Additionally, [Avalonia does not support Windows 7 or earlier](https://github.com/AvaloniaUI/Avalonia/wiki/Runtime-Requirements) and support for these older OSes is a primary use-case for PhotoDemon.
